Pomodoro.Services/StatisticsService.cs

```diff
@@ -1,110 +1,51 @@
-
-				monthlyStatistics.TasksCompleted = tasks.Count();
-
-				foreach (var pomo in tasks)
-				{
-					monthlyStatistics.PomodorosDone += pomo.PomodorosCount;
-				}
-				foreach (var time in tasks)
-				{
-					monthlyStatistics.TimeSpent += time.TimeSpent;
-				}
-			}
-
-			return await Task.FromResult(monthlyStatistics);
-
-		}
-
-		public async Task<AnnualStatistics> GetAnnualStatisticsAsync(Guid userId, int year)
-		{
-			AnnualStatistics annualStatistics = new AnnualStatistics();
-
-			annualStatistics.UserId = userId;
-			annualStatistics.Year = year;
-
-			using (AppDbContext db = new AppDbContext())
-			{
-				IQueryable<Completed> monthsTasks = db.CompletedTasks;
-
-				foreach (int m in Enum.GetValues(typeof(Month)))
-				{
-					monthsTasks = monthsTasks.Where(t => t.ActualDate.Year == annualStatistics.Year);
-					monthsTasks = monthsTasks.Where(t => t.ActualDate.Month == m);
-
-					AnalyticsPerMonth temp = new AnalyticsPerMonth();
-					
-					temp.Month = (Month)m;
-					foreach (var d in monthsTasks)
-					{
-						temp.PomodorosDone += d.PomodorosCount;
-					}
-					annualStatistics.AnalyticsPerMonths?.Add(temp);
-				}
-
-			}
-
-			return await Task.FromResult(annualStatistics);
-
-		}
-	}
+    public class StatisticsService:IStatisticsService
+    {
+        private readonly IMapper _mapper;
+        private readonly ICompletedRepository _completedRepository;
+
+        public StatisticsService(IMapper mapper, ICompletedRepository repository)
+        {
+            this._mapper = mapper;
+            this._completedRepository = repository;
+        }
+        
+        public async Task<DailyStatistics> GetDailyStatisticsAsync(Guid userId, DateOnly day)
+        {
+            var dailystatistics = await _completedRepository.FindAsync(
+                u=>u.Task != null &&
+                   u.Task.UserId == userId &&
+                   u.ActualDate.Year == day.Year &&
+                   u.ActualDate.Month == day.Month && 
+                   u.ActualDate.Day == day.Day
+            );
+            return  _mapper.Map<DailyStatistics>(dailystatistics);
+        }
+
+        public async Task<MonthlyStatistics> GetMonthlyStatisticsAsync(Guid userId, int year, int month)
+        {
+            var monthlystatistics = await _completedRepository.FindAsync(
+                m=> m.Task != null &&
+                    m.Task.UserId == userId && 
+                    m.ActualDate.Year == year && 
+                    m.ActualDate.Month == month );
+
+            return _mapper.Map<MonthlyStatistics>(monthlystatistics);
+        }
+
+        public async Task<AnnualStatistics> GetAnnualStatisticsAsync(Guid userId, int year)
+        {
+            var annualstatistics = await _completedRepository.FindAsync(
+                a => a.Task != null &&
+                     a.Task.UserId == userId && 
+                     a.ActualDate.Year == year
+            );
+            return _mapper.Map<AnnualStatistics>(annualstatistics);
```

Comment:
Unfortunately, it's not going to work. And both of the previous methods, too. Because you haven't provided any configs for AutoMapper. Thus, this library doesn't know how to map from `IEnumerable<Completed>` to `AnnualStatistics`, for example. Besides, since statistics generation is part of the business logic, I'm not sure if AutoMapper should do this work at all.